### PR TITLE
Dont pass transaction in URL in multi signature continuous flow

### LINKED ISSUE #5374: Dont pass transaction in URL in multi signature continuous flow
### Description

When switching account in a continuous multi sig flow we currently pass ([This PR](https://github.com/LiskHQ/lisk-desktop/pull/5366)) the transaction url as a url params.

It would be better if we use:
1. Url state (Dont think it will work since we link to a modal)
2. Redux for this temporary state

### Motivation

- Current way creates a messy URL and could probably cause problems when a lot of multi signature accounts since url params have a limit of 2048 character.

### PR DESCRIPTION
**Pull request by oskarleonard**
### What was the problem?

This PR resolves #5374 #5388

### How was it solved?

- Moving transaction string to url state since its max size is 640k characters compared to url params max size of 2k characters.
- Since URL is no longer unique between certain switches i also had to add a `generateUniqueId` in the url.
- Check for required remaining members to determine if a transaction is complete

### How was it tested?
Because of issue #5388 we unfortunately need to test all edge cases again to prevent regression in another state.
So please test the cases from [this PR](https://github.com/LiskHQ/lisk-desktop/pull/5366)
As well as the issue raised in this issue  #5388

### CODE REVIEW

File: src/modules/account/hooks/useCurrentAccount.js [hunk 1]
```diff
@@ -40,7 +41,17 @@ export function useCurrentAccount() {
     } else {
       dispatch(setCurrentAccount(encryptedAccount));
       if (redirect) {
-        history.push(referrer || routes.wallet.path);
+        if (urlState) {
+          const relativeUrlPath = referrer || routes.wallet.path;
+          const { pathname, search } = new URL(relativeUrlPath, window.location.origin);
```
Comment (ikem-legend) [inline src/modules/account/hooks/useCurrentAccount.js]: Since this same variable is used in line 53, it can be defined before the conditions then used in each condition. The current implementation is also fine

**Review by ikem-legend — DISMISSED** [review]

File: src/modules/transaction/components/Multisignature/Multisignature.js [hunk 2]
```diff
@@ -28,15 +29,18 @@ export const PartiallySignedActions = ({
...
-  const [, setCurrentAccount] = useCurrentAccount();
+  const [currentAccount, setCurrentAccount] = useCurrentAccount();
 
   const handleSwitchAccount = () => {
     const stringifiedTransaction = encodeURIComponent(JSON.stringify(transactionJSON));
+    const uniqueUrlIdToPreventHashError = generateUniqueId();
     setCurrentAccount(
       nextAccountToSign,
-      `/wallet?modal=signMultiSignTransaction&stringifiedTransaction=${stringifiedTransaction}`,
+      `/wallet?modal=signMultiSignTransaction&prevAccount=${currentAccount.metadata.address}&uniqueId=${uniqueUrlIdToPreventHashError}`,
```
Comment (eniolam1000752) [inline src/modules/transaction/components/Multisignature/Multisignature.js:39]: I don't see where uniqueId and prevAccount is consued

**Review by eniolam1000752 — DISMISSED** [review]

File: src/modules/transaction/components/Multisignature/Multisignature.js (hunk 2 above)
Comment (oskarleonard) [reply src/modules/transaction/components/Multisignature/Multisignature.js:39]: They are used for the URL, not inside the app, it was added to make the url unique because otherwise there were some strange bugs, which is why i named it `uniqueUrlIdToPreventHashError`. Got a warning as well in chrome console in one of the cases when i didnt add them (dont remember which one of all cases i tested). I think it was, if you use react router hash and push two identical urls to the history stack it have some unintended consequences regarding the url state... 

Probably `prevAccount` is not needed now since url should be unique with the help of `generateUniqueId()` but i didnt want to remove it and have to test all cases again for that change. Especially since we might want to use it in the future (in another issue) when we will allow for switching back to the starting account

File: src/modules/account/hooks/useCurrentAccount.js (hunk 1 above)
Comment (oskarleonard) [reply src/modules/account/hooks/useCurrentAccount.js]: Make sense, fixed in latest commit

File: src/modules/transaction/components/Multisignature/Multisignature.js (hunk 2 above)
Comment (eniolam1000752) [reply src/modules/transaction/components/Multisignature/Multisignature.js:39]: okay cool

**Review by eniolam1000752 — APPROVED** [review]

**Review by ikem-legend — APPROVED** [review]
